workspaces/jenkins/plugins/scaffolder-backend-module-jenkins/src/actions/job/create.ts

```diff
@@ -0,0 +1,158 @@
+  }
+  url = url.concat(`/createItem?name=${jobName}`);
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      Authorization: `Basic ${token}`,
+      'Content-Type': 'application/xml',
+    },
+    body: jobXml,
+  });
+  if (response.ok) {
+    ctx.logger.info(`Successfully created job ${jobName}`);
+    ctx.logger.info(
+      `Job URL: ${server}${
+        folderName ? `/job/${folderName}` : ''
+      }/job/${jobName}`,
+    );
+  }
+  if (!response.ok) {
+    throw new Error(
+      `Failed to create job, ${response.status} ${response.statusText}`,
+    );
+  }
+}
+
+/**
+ * This createJob function, creates a job given a job name and own configuration file as xml format
+ *
+ * @param jenkins - The client to interact with jenkins instance
+ * @returns Empty response, in case of error an exception will be thrown by jenkins client
+ */
+export function createJob(jenkins: Jenkins, config: RootConfigService) {
+  return createTemplateAction<{
+    jobName: string;
+    jobXml: string;
+    folderName: string;
+    configPath: string;
+    serverUrl: string;
+  }>({
+    id: 'jenkins:job:create',
+    description: 'Create a job jenkins given a name and gitlab repo',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['jobName'],
+        properties: {
+          jobName: {
+            title: 'Jenkins job name',
+            description: 'Name of jenkins item',
+            type: 'string',
+          },
+          jobXml: {
+            title: 'Jenkins job xml',
+            description: 'XML of job used by jenkins to create the job',
+            type: 'string',
+          },
+          folderName: {
+            type: 'string',
+            title: 'Folder of jobName',
+          },
+          configPath: {
+            type: 'string',
+            title: 'Path to config file of job',
+          },
+          serverUrl: {
+            type: 'string',
+            title: 'URL',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      ctx.logger.info(`Creating jenkins job ${ctx.input.jobName}`);
+
+      ctx.logger.debug(
+        'Trying to create job jenkins with this xml {}',
+        ctx.input.jobXml,
+      );
+      const { configPath, folderName, serverUrl } = ctx.input;
+
+      let jobXml = ctx.input.jobXml;
+      if (configPath) {
+        jobXml = await fs.readFile(
+          resolveSafeChildPath(ctx.workspacePath, configPath),
+          'utf8',
+        );
+      }
+
+      let jobName;
+      if (folderName) {
+        jobName = `${folderName}/${ctx.input.jobName}`;
+      } else {
+        jobName = ctx.input.jobName;
+      }
+
+      if (!jobXml) {
+        throw new Error(
+          'JobXml cannot be null or empty, please configure with inline content or from xml file!',
+        );
+      }
+
+      if (serverUrl) {
+        await createJobByServerUrl(config, ctx, jobXml);
+        return;
+      }
+
+      await jenkins.job.create(jobName, jobXml);
```

Comment:
Otherwise, an alternative to re-using the old logic could have been:

```suggestion
      const client = serverUrl ? buildJenkinsClient(config, serverUrl) : jenkins;
      await client.job.create(jobName, jobXml);
```

Reply:
i think is the best way